workloads/kube-burner-ocp-wrapper/run.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash -e
+
+set -e
+
+hypershift(){
+  if [[ -z ${MC_KUBECONFIG} ]]; then
+    echo "MC_KUBECONFIG variable must be defined"
+    exit 1
+  fi
+  
+  HC_ID=$(oc get infrastructure cluster -o go-template --template='{{.status.infrastructureName}}')
+  HC_NAME=$(oc get infrastructure cluster -o go-template --template='{{range .status.platformStatus.aws.resourceTags}}{{if eq .key "api.openshift.com/name" }}{{.value}}{{end}}{{end}}')
+  
+  
+  if [[ -z ${HC_ID} ]] || [[ -z ${HC_NAME} ]]; then
+    echo "Couldn't obtain hosted cluster id and/or hosted cluster name"
+    echo -e "HC_ID: ${HC_ID}\nHC_NAME: ${HC_NAME}"
+    exit 1
+  fi
+  
+  # Hosted control-plane namespace is composed by the cluster ID plus the cluster name
+  HCP_NAMESPACE=${HC_ID}-${HC_NAME}
+  
+  echo "Creating OBO route"
+  oc --kubeconfig=${MC_KUBECONFIG} apply -f - << EOF
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app.kubernetes.io/instance: hypershift-monitoring-stack
+  name: prometheus-hypershift-monitoring-stack
+  namespace: openshift-observability-operator
+spec:
+  ports:
+  - port: 9090
+    protocol: TCP
+    targetPort: web
+  selector:
+    app.kubernetes.io/instance: hypershift-monitoring-stack
+---
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: prometheus-hypershift
+  namespace: openshift-observability-operator
+spec:
+  port:
+    targetPort: 9090
+  to:
+    name: prometheus-hypershift-monitoring-stack
+EOF
+  echo "Fetching OBO endpoint"
+  MC_OBO=http://$(oc --kubeconfig=${MC_KUBECONFIG} get route -n openshift-observability-operator prometheus-hypershift -o jsonpath="{.spec.host}")
+  MC_PROMETHEUS=https://$(oc --kubeconfig=${MC_KUBECONFIG} get route -n openshift-monitoring prometheus-k8s -o jsonpath="{.spec.host}")
+  MC_PROMETHEUS_TOKEN=$(oc --kubeconfig=${MC_KUBECONFIG} sa new-token -n openshift-monitoring prometheus-k8s)
+  HOSTED_PROMETHEUS=https://$(oc get route -n openshift-monitoring prometheus-k8s -o jsonpath="{.spec.host}")
+  HOSTED_PROMETHEUS_TOKEN=$(oc sa new-token -n openshift-monitoring prometheus-k8s)
+  echo "Exporting required vars"
+  cat << EOF
+MC_OBO: ${MC_OBO}
+MC_PROMETHEUS: ${MC_PROMETHEUS}
+MC_PROMETHEUS_TOKEN: ${MC_PROMETHEUS_TOKEN}
+HOSTED_PROMETHEUS: ${HOSTED_PROMETHEUS}
+HOSTED_PROMETHEUS_TOKEN: ${HOSTED_PROMETHEUS_TOKEN}
+EOF
+  export MC_OBO MC_PROMETHEUS MC_PROMETHEUS_TOKEN HOSTED_PROMETHEUS HOSTED_PROMETHEUS_TOKEN HCP_NAMESPACE
+}
+
+if [[ -z ${WORKLOAD} ]]; then
+  echo "WORKLOAD variable must be defined"
+  exit 1
+fi
+if [[ ${HYPERSHIFT} == "true" ]]; then
```

Comment:
Just to make sure if this block is not redundant, How is this if block different from the logic being executed in the if block at line84?

Reply:
Should be fixed now